discord/user.py

```diff
@@ -383,6 +391,24 @@ def blocked(self):
 
         http = self._state.http
 
+        try:
+            house = fields['house']
+            if house is None:
+                await http.leave_hypesquad_house()
+
+            elif 'first_name' not in fields or 'last_name' not in fields:
+                raise ClientException('When changing hypesquad houses first_name and last_name fields are required.')
+
+            else:
+                if isinstance(house, HypesquadHouse):
+                    value = house.value
+                else:
+                    value = int(house)
+
+                await http.change_hypesquad_house(value, fields['first_name'], fields['last_name'])
+        except KeyError:
```

Comment:
Can't return early, we still have to send out a request to edit the ClientUser. can't bring back the except farther too, since we'd have a NameError which that entire block inside the try except relies on `house` being present and a valid house type.

This block is supposed to simply fail silently if we get no house in the kwargs, it's not supposed to affect the success of the regular edit call